### PR TITLE
Bugfix/mutation query weirdness

### LINKED ISSUE #109: gene auto-complete in mutations is kind of wacky
type in `GRIN` - suggests CAMK4 ACSBG1 GSR NR3C1 DCAF6
type in `GRIN2` - suggests GPRIN2
type in `GRIN2A` - suggests GRIN2A GRIN2D GRIN2B

### PR DESCRIPTION
**Pull request by ramenhog**
## Issue Number
Closes #109 

## Purpose/Implementation Notes
* confirmed that search responses were accurate/expected
* updated search results to be "No results found" if no results were found

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Functional tests
* when query endpoint returns no results, UI shows that there's no results

## Checklist

_Put an `x` in the boxes that apply._

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/js/components/queryBuilder/queryParamSelector/queryParamSelector.tpl.html [hunk 1]
```diff
@@ -42,7 +42,8 @@ <h2>{{$ctrl.currentState()}} Selector</h2>
 		is-searching="$ctrl.isSearching"
 		is-filter-disabled="$ctrl.currentState() === 'mutations'"
 	></cg-table>
-	<p ng-if="!$ctrl.searchResults.length && $ctrl.currentState() === 'mutations' && !$ctrl.isSearching">Begin searching to see results</p>
+	<p ng-if="!$ctrl.searchQuery.length && $ctrl.currentState() === 'mutations' && !$ctrl.isSearching">Begin searching to see results.</p>
+	<p ng-if="$ctrl.searchQuery.length && !$ctrl.searchResults.length && $ctrl.currentState() === 'mutations' && !$ctrl.isSearching">No results found.</p>
```
Comment (dongbohu) [inline app/js/components/queryBuilder/queryParamSelector/queryParamSelector.tpl.html]: These three conditions will have 8 states totally, two of which are handled here by `ng-if`. What about the other 6 states? We assume that they are all digested somewhere by the code?

Comment (ramenhog) [reply app/js/components/queryBuilder/queryParamSelector/queryParamSelector.tpl.html]: Which six states specifically?

Comment (dongbohu) [reply app/js/components/queryBuilder/queryParamSelector/queryParamSelector.tpl.html]: Here is what I meant: The code includes three boolean values:
(1) $ctrl.searchQuery.length === 0
(2) $ctrl.currentState() === 'mutations' 
(3) $ctrl.isSearching
Each boolean has two states (true or false), so the combination of these three will have 2x2x2 = 8 states totally. The two `ng-if` are handling 2 of them. What about the other 6 states?

Comment (ramenhog) [reply app/js/components/queryBuilder/queryParamSelector/queryParamSelector.tpl.html]: Yes, the other states are being digested by the code. 

When the current state is diseases, it automatically loads all the results and doesn't need to worry about the length of the search query (the user doesn't need to be prompted to start searching). And the `cgTable` component also has an `ng-if` on the "isSearching" boolean and only appears after that condition is `false`.

**Pull request merged**